tests/v2/validation/provisioning/rke1/support_matrix_test.go

```diff
@@ -0,0 +1,178 @@
+	cnis               []string
+	nodeProviders      []string
+	advancedOptions    provisioning.AdvancedOptions
+}
+
+func (s *SupportMatrixTestSuite) TearDownSuite() {
+	s.session.Cleanup()
+}
+
+func (s *SupportMatrixTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	s.session = testSession
+
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+
+	s.kubernetesVersions = clustersConfig.RKE1KubernetesVersions
+	s.cnis = clustersConfig.CNIs
+	s.nodeProviders = clustersConfig.NodeProviders
+	s.advancedOptions = clustersConfig.AdvancedOptions
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(s.T(), err)
+
+	s.client = client
+
+	s.kubernetesVersions, err = kubernetesversions.Default(s.client, clusters.RKE1ClusterType.String(), s.kubernetesVersions)
+	require.NoError(s.T(), err)
+
+	enabled := true
+	var testuser = namegen.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(s.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(s.T(), err)
+
+	s.standardUserClient = standardUserClient
+}
+
+func (s *SupportMatrixTestSuite) TestSupportMatrixChecks() {
+	require.GreaterOrEqual(s.T(), len(s.cnis), 1)
+
+	nodeRoles := []string{
+		"--etcd --controlplane --worker",
+	}
+
+	tests := []struct {
+		name      string
+		nodeRoles []string
+		client    *rancher.Client
+	}{
+		{"1 Node all roles " + provisioning.StandardClientName.String(), nodeRoles, s.standardUserClient},
+	}
+	var name string
+	for _, tt := range tests {
+		testSession := session.NewSession()
+		defer testSession.Cleanup()
+
+		client, err := tt.client.WithSession(testSession)
+		require.NoError(s.T(), err)
+
+		for _, nodeProviderName := range s.nodeProviders {
+			externalNodeProvider := provisioning.ExternalNodeProviderSetup(nodeProviderName)
+			providerName := " Node Provider: " + nodeProviderName
+			for _, kubeVersion := range s.kubernetesVersions {
+				for _, cni := range s.cnis {
+					name = tt.name + providerName + " Kubernetes version: " + kubeVersion + " cni: " + cni
+					s.Run(name, func() {
+						cluster, err := provisioningRKE1CustomCluster(s.T(), client, externalNodeProvider, tt.nodeRoles, kubeVersion, cni, s.advancedOptions)
+						require.NoError(s.T(), err)
+
+						err = matrix.CheckSupportMatrixImageVersions(client, cluster.ID)
+						require.NoError(s.T(), err)
+					})
+				}
+			}
+		}
+	}
+}
+
+func provisioningRKE1CustomCluster(t *testing.T, client *rancher.Client, externalNodeProvider provisioning.ExternalNodeProvider, nodesAndRoles []string, kubeVersion, cni string, advancedOptions provisioning.AdvancedOptions) (*management.Cluster, error) {
```

Comment:
I'm not sure, but does this need to live in its own file since it is a helper function?

Reply:
If you want, we can, no big deal. I didn't initially to just keep the file to one since this will only be used for support matrix checks and nothing else.

Reply:
I see, it should be fine then.

Reply:
Does this need to be it's own function? Why not use the existing one?

Reply:
So this function specifically does a check for the `etcd` version. Additionally, since this is specifically performing our RKE1 support matrix checks, we don't need to run all the validation checks as we do in `custom_cluster.go`. Since that's the case, I just made this private function and cut down some of the code.

Reply:
I don't agree Markus, that checks we make in the existing custom cluster func, are minimal and don't take much time at all. I would argue that is the preferred stated of a custom rke1 cluster. I think Ugur's idea is good, we just add them as separate tests within its own `.Run()` in the existing provisioning suite.

Reply:
@slickwarren what do you think of Ugur's comment above? To give further context, as support matrix RKE1 validations occur once every release, we didn't want to have these additional validations run on a regular basis.

Because of that, that's why the implementation is how it is. I'm not totally against putting these checks in the custom cluster suite, but I would prefer to not do so since the support matrix checks are not on a regular basis, if that makes sense?

Reply:
If it's in it's own .Run() they wouldn't run all the time. It would be similar to how we do scaling for RKE1

Reply:
Oh okay. If we go that route, I probably would want to update the README to explain the support matrix checks inclusion then. I will look into that.

Reply:
Pushed a new commit that resolves this concern.